java/springmvc/src/main/java/com/lacunasoftware/pkisuite/controller/AuthenticationCloudExpressController.java

```diff
@@ -0,0 +1,182 @@
+package com.lacunasoftware.pkisuite.controller;
+
+import com.lacunasoftware.pkiexpress.*;
+import com.lacunasoftware.pkisuite.model.express.AuthenticationCloudExpressModel;
+import com.lacunasoftware.pkisuite.util.Util;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.ModelAndView;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.List;
+
+
+/**
+ * This sample is responsible to perform a OAuth flow to communicate with PSCs to perform a
+ * signature. To perform this sample it's necessary to configure PKI Express with the credentials of
+ * the services by executing the following sample:
+ *
+ *    pkie config --set trustServices:<provider>:<configuration>
+ *
+ * All standard providers:
+ *    - BirdId
+ *    - ViDaaS
+ *    - NeoId
+ *    - RemoteId
+ *    - SafeId
+ * It's possible to create a custom provider if necessary.
+ *
+ * All configuration available:
+ *    - clientId
+ *    - clientSecret
+ *    - endpoint
+ *    - provider
+ *    - badgeUrl
+ *    - protocolVariant (error handling, normally it depends on the used provider)
+ *
+ * This sample will only show the PSCs that are configured.
+ */
+@Controller
+public class AuthenticationCloudExpressController {
+
+	// Redirect URL where it's accessed after OAuth flow is finished.
+	private final String REDIRECT_URL = "http://localhost:60695/authentication-cloud-express/complete";
+
+	/**
+	 * This action will render a page that request a CPF to the user. This CPF is used to discover
+	 * which PSCs have a certificate containing that CPF.
+	 */
+	@GetMapping("/authentication-cloud-express")
+	public ModelAndView get(
+	) {
+		return new ModelAndView("/authentication-cloud-express/index");
+	}
+
+    /**
+	 * This action will be called after the user press the button "Search" on index page. It will
+	 * search for all PSCs that have a certificate with the provided CPF. Thus, it will start the
+	 * authentication process and return a URL to redirect the user to perform the authentication.
+	 *
+	 * After this action the user will be redirected, and to store the local data (fileId) to be user
+	 * after the user returns to your application. We use the parameter "customState", the last
+	 * parameter of the method discoverByCpfAndStartAuth(). This parameter will be recovered in the
+	 * next action.
+	 */
+	@PostMapping("/authentication-cloud-express/discover")
+	public ModelAndView discover(
+		@RequestParam(value="cpf") String cpf
+	) throws IOException {
+        // Get an instance of the Authentication class, responsible for contacting PKI Express to
+		// start and complete the authentication operation.
+        Authentication auth = new Authentication();
```

Comment:
Can you fix the indentation of the flies? It seems to have an mixture of tab and spaces.

Reply:
Made the changes in indentation in the next commits